src/server/game/Entities/Player/Player.cpp

```diff
@@ -6383,6 +6383,54 @@ bool Player::UpdateGatherSkill(uint32 SkillId, uint32 SkillValue, uint32 RedLeve
     return false;
 }
 
+float getProbabilityOfLevelUp(uint32 SkillValue)
+{
+    /* According to El's Extreme Angling page, from 1 to 115 the probability of a skill
+     * level up is 100% since 100/1 = 100. From 115 - 135 should average 2 catches per
+     * skill up so that means 100/2 = 50%. The if statement returns the probability
+     * depending on the player's SkillValue.
+     */
+    if (SkillValue > 115 && SkillValue <= 135)
```

Comment:
I am not a c++ pro so just throwing out thoughts.
Could this many checks on a larger scale case minor delays?
Would it be better to use a switch case?

Reply:
I understand the worry, I don't think this should cause any delay. However, I haven't found a better solution so far. A switch is a bad option since you would need a case for every 450 numbers in a skill. That I believe would cause a minor delay.

I'm open to suggestions! I thought about using a query but I think it's so silly to query the DB every time someone catches something. It could cause server hiccups.

Reply:
Why would you need 450 cases? Isn't it possible to use case 0 to return 0.0f and any other case just uses the calculation with the arrays?

Reply:
The code has changed Hacki, ditched this logic for a cleaner one